File: prince-of-versions/src/main/java/co/infinum/princeofversions/PrinceOfVersionsCallbackExecutor.java
package co.infinum.princeofversions;

import android.os.Handler;
import android.os.Looper;

import java.util.concurrent.Executor;
import javax.annotation.Nonnull;

/**
 * Implementation of {@link Executor} which runs task on main thread.
 */
public class PrinceOfVersionsCallbackExecutor implements Executor {

    private final Handler handler = new Handler(Looper.myLooper());

    @Override
    public void execute(@Nonnull final Runnable command) {
        handler.post(command);
    }
}

File: prince-of-versions/src/main/java/co/infinum/princeofversions/PrinceOfVersionsCallbackExecutor.kt
package co.infinum.princeofversions

import android.os.Handler
import android.os.Looper
import java.util.concurrent.Executor

/**
 * An implementation of [Executor] that runs tasks on the Android main thread.
 */
class PrinceOfVersionsCallbackExecutor : Executor {

    private val handler = Handler(Looper.getMainLooper())

    override fun execute(command: Runnable) {
        handler.post(command)
    }
}